pkg/cleaner/cleaner_test.go

```diff
@@ -0,0 +1,118 @@
+package cleaner
```

Comment:
@burmanm @jsanda 
Providing a [link](https://github.com/k8ssandra/k8ssandra/tree/operator-integration-tests) over to the integration testing foundations underway.   

As part of those iterations, there are quite a few lessons-learned and functions available that we could reference for much of this cleanup.

I was hopeful we could get that PR merged before we got too far in the other areas.  As such, I would be willing to assist with a ticket to get that effort merged with this cleanup test as we move forward.

Reply:
I'm not following really, what would this benefit from? It only creates two CassandraDatacenters to the testenv and then runs the cleaner basically.

Reply:
The thought is we would have more sameness/consistency in our k8ssandra tests with build-up and reuse of integration test functions (e.g. util module). Maybe something similar for our unit testing.

[Here](https://github.com/k8ssandra/k8ssandra/blob/operator-integration-tests/tests/integration/util/lib.go) is the current version of the `util module` where I've been building some common operations.

I included some setup & helper models as well to manage the various options (operator, cluster, network, etc.) that  many of the integration tests require.

Reply:
Sure, but none of those are really useful for this test. This test isn't running against real cluster.

Reply:
@jeffbanks The test use [envtest](https://github.com/kubernetes-sigs/controller-runtime/tree/master/pkg/envtest). As @burmanm, it is not a real cluster. envtest does deploy etcd and the apiserver, but that is it.